CosmosApiChangeFeedForMessageRetention/handler.ts

```diff
@@ -0,0 +1,145 @@
+import * as TE from "fp-ts/lib/TaskEither";
+import * as RA from "fp-ts/lib/ReadonlyArray";
+import * as O from "fp-ts/lib/Option";
+
+import { flow, pipe } from "fp-ts/lib/function";
+import {
+  MessageStatus,
+  MessageStatusModel,
+  RetrievedMessageStatus
+} from "@pagopa/io-functions-commons/dist/src/models/message_status";
+import { Context } from "@azure/functions";
+import {
+  Profile,
+  ProfileModel
+} from "@pagopa/io-functions-commons/dist/src/models/profile";
+import { RejectedMessageStatusValueEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/RejectedMessageStatusValue";
+import { MessageModel } from "@pagopa/io-functions-commons/dist/src/models/message";
+import { FiscalCode } from "@pagopa/io-functions-commons/dist/generated/definitions/FiscalCode";
+import { CosmosErrors } from "@pagopa/io-functions-commons/dist/src/utils/cosmosdb_model";
+import { Ttl } from "@pagopa/io-functions-commons/dist/src/utils/cosmosdb_model_ttl";
+
+/**
+  the timestamp related to 2022-11-23   20:00:00
+  we have released the version of fn-service that adds the TTL to message and message-status,
+  so we do no longer want to clean up after this ts
+ */
+export const RELEASE_TIMESTAMP = 1669233600;
+
+/**
+  3 years in seconds
+ */
+
+export const TTL_VALUE = 94670856 as Ttl;
+
+export const isStatusRejected = (document: RetrievedMessageStatus): boolean =>
+  document.status === RejectedMessageStatusValueEnum.REJECTED;
+
+/** Return true if the given resource does NOT have the ttl */
+export const hasNotTTL = (document: RetrievedMessageStatus): boolean =>
+  document.ttl ? false : true;
+
+/**
+  Given 2 timestamps return true whether the first one is before the second
+ */
+
+export const isBeforeDate = (firstTs: number, secondTs: number): boolean =>
+  firstTs < secondTs;
+
+export const isEligibleForTTL = (
+  document: RetrievedMessageStatus
+): TE.TaskEither<string, RetrievedMessageStatus> =>
+  pipe(
+    document,
+    TE.fromPredicate(
+      isStatusRejected,
+      () => `This message status is not rejected`
+    ),
+    TE.chain(
+      TE.fromPredicate(
+        // eslint-disable-next-line no-underscore-dangle
+        () => isBeforeDate(document._ts, RELEASE_TIMESTAMP),
+        () =>
+          // eslint-disable-next-line no-underscore-dangle
+          `the timestamp of the document ${document.id} (${document._ts}) is after the RELEASE_TIMESTAMP ${RELEASE_TIMESTAMP}`
```

Comment:
Could we add an event to let us know when we reached message status after `RELEASE_TIMESTAMP`?